src/eei.cpp

```diff
@@ -742,17 +742,8 @@ inline int64_t maxCallGas(int64_t gas) {
 
       call_message.gas = gas;
 
-      // FIXME: consider refactoring this
       if (import->base == Name("call") || import->base == Name("callCode")) {
-        try {
-          ensureCondition(msg.depth < 1024, OutOfGas, "Call depth exceeded.");
-
-          ensureSenderBalance(call_message.value);
-        } catch (OutOfGas const&) {
-          // An out of gas error here doesn't really mean we're out of gas, it just means
-          // we don't have sufficient gas for the call value. We should return
-          // an error rather than throwing.
-          //
+        if (!enoughSenderBalanceFor(call_message.value) || (msg.depth >= 1024)) {
```

Comment:
You should check the depth first, because it's quicker check.

Reply:
Right. I think actually we need to be more specific consensus wise and define which one is first.

Reply:
The order does not affect EVM consensus because in any case the same failure is returned.

Reply:
In current EVM it is not an issue, but I think it is bad design not being specific of order.